tests/test_language_negotiation.py

```diff
@@ -36,33 +36,77 @@ def test_language_negotiation_accept_header_contains_languages_with_q_factor() -
+def test_language_negotiation__q_value_below_0_0() -> None:
+    """Should be ignored and the default language returned."""
+    accept_language_header: List[str] = ["nb-NO;q=-1.0"]
+    content_language = decide_language(accept_language_header, SUPPORTED_LANGUAGES)
+    assert (
+        SUPPORTED_LANGUAGES[0] == content_language  # en-GB
+    ), f"For header-value '{accept_language_header}', content-language should be {SUPPORTED_LANGUAGES[0]}."  # noqa: B950
+
+
+def test_language_negotiation_accept_language_header_contains_languages_and_no_supported_language() -> None:  # noqa: B950
     """Should raise NoAgreeableLanguageError."""
     accept_language_header: List[str] = ["en-GB", "nb-NO"]
     supported_languages: List[str] = []
     with pytest.raises(NoAgreeableLanguageError):
         decide_language(accept_language_header, supported_languages)
 
 
-def test_language_negotiation_accept_header_contains_only_semicolon_and_q() -> None:  # noqa: B950
+def test_language_negotiation_accept_language_header_contains_only_semicolon_and_q() -> None:  # noqa: B950
     """Should raise NoAgreeableLanguageError."""
     accept_language_header: List[str] = [";q=0.8"]
     supported_languages: List[str] = []
     with pytest.raises(NoAgreeableLanguageError):
         decide_language(accept_language_header, supported_languages)
+
+
+def test_language_negotiation_accept_language_header_contains_only_star() -> None:
+    """Should return the default language."""
+    accept_language_header: List[str] = ["*"]
+    content_language = decide_language(accept_language_header, SUPPORTED_LANGUAGES)
+    assert (
+        SUPPORTED_LANGUAGES[0] == content_language  # "en-GB"
+    ), f"For header-value '{accept_language_header}', content-language should be {SUPPORTED_LANGUAGES[0]}."  # noqa: B950
+
+
+def test_language_negotiation_accept_language_header_contains_star_with_highest_q() -> None:
+    """Should return the default language."""
+    accept_language_header: List[str] = ["nb-NO;q=0.8,*;q=0.9"]
+    content_language = decide_language(accept_language_header, SUPPORTED_LANGUAGES)
+    assert (
+        SUPPORTED_LANGUAGES[0] == content_language  # "en-GB"
+    ), f"For header-value '{accept_language_header}', content-language should be {SUPPORTED_LANGUAGES[0]}."  # noqa: B950
+
+
+def test_language_negotiation_accept_language_header_empty() -> None:
+    """Should return the default language."""
+    accept_language_header: List[str] = []
+    content_language = decide_language(accept_language_header, SUPPORTED_LANGUAGES)
+    assert (
+        SUPPORTED_LANGUAGES[0] == content_language  # "en-GB"
+    ), f"For header-value '{accept_language_header}', content-language should be {SUPPORTED_LANGUAGES[0]}."  # noqa: B950
+
+
+def test_language_negotiation_accept_language_header_contains_only_empty_string() -> None:
+    """Should raise NoAgreeableLanguageError."""
+    accept_language_header: List[str] = [""]
+    supported_languages: List[str] = []
```

Comment:
Må da ha noen supported for at testen skal gi verdi? Blir det ikke alltid NoAgreeableLanguageError om den er tom?

Reply:
Fiksa i 6223a859e091382af3c89852df3c2189b6ad488b